bfdriver/src/platform/efi/src/boot.c

```diff
@@ -0,0 +1,98 @@
+//
+// Bareflank Hypervisor
+// Copyright (C) 2018 Assured Information Security, Inc.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+#include "bfefi.h"
+#include "bflib.h"
+#include "bfloader.h"
+
+EFI_STATUS bf_boot_next_by_order()
+{
+    EFI_GUID global_guid = EFI_GLOBAL_VARIABLE;
+
+    EFI_STATUS status;
+    EFI_LOADED_IMAGE *li;
+    status = gBS->HandleProtocol(this_image_h,
+                                 &gEfiLoadedImageProtocolGuid,
+                                 (VOID **)&li);
+    //bf_dump_hex(0,0,li->LoadOptionsSize,li->LoadOptions);
```

Comment:
We have a debug switch 'NDEBUG' defined in bfconstants.h. We should use that here or create an EFI-specific debug switch that turns on print statements